notion_integration/notion_book_database.py

```diff
@@ -1,110 +1,138 @@
+        item_retrieved = response["results"][0]["properties"]
 
-        #format date to human readable
-        date_string = item_retrieved['Date']['date']['start']
-        date_format = '%Y-%m-%dT%H:%M:%S.%f%z'
+        # format date to human readable
+        date_string = item_retrieved["Date"]["date"]["start"]
+        date_format = "%Y-%m-%dT%H:%M:%S.%f%z"
         datetime_obj = datetime.strptime(date_string, date_format)
-        date_formatted = datetime_obj.strftime('%Y-%m-%d %H:%M')
+        date_formatted = datetime_obj.strftime("%Y-%m-%d %H:%M")
 
-        #format book to human readable
-        book_unformatted = item_retrieved['Book']['rich_text'][0]['text']['content']
-        book_formatted = book_unformatted.replace('_',' ')
+        # format book to human readable
+        book_unformatted = item_retrieved["Book"]["rich_text"][0]["text"]["content"]
+        book_formatted = book_unformatted.replace("_", " ")
 
-        #format location
-        location_unformatted = item_retrieved['Location']['number']
+        # format location
+        location_unformatted = item_retrieved["Location"]["number"]
         location_formatted = str(location_unformatted)
 
-        row_dict = {
-            'Highlight': item_retrieved['Highlight']['rich_text'][0]['text']['content'],
-            'Book': book_formatted,
-            'Author': item_retrieved['Author']['rich_text'][0]['text']['content'],
-            'Location': location_formatted,
-            'Date': date_formatted,
-            'Note': item_retrieved['Note']['title'][0]['text']['content'],
-        }
+        thought = WiseThought(
+            highlight=item_retrieved["Highlight"]["rich_text"][0]["text"]["content"],
+            title=book_formatted,
+            author=item_retrieved["Author"]["rich_text"][0]["text"]["content"],
+            location=location_formatted,
+            date=date_formatted,
+            note=item_retrieved["Note"]["title"][0]["text"]["content"],
+        )
 
-        return row_dict
+        return thought
 
     def get_random_row_data(self):
-        response = self.query_database_notion_for_random_row()
-        data = self.retrieve_content_from_response(response)
-        return data
-    
-    def send_to_notion(self, data):
+        """This function is used to retrieve the random row from the notion database and
+        return it as WiseThought object"""
+        response = self._retrieve_random_row_from_database()
+        thought = self._extract_thought_from_response(response)
+        return thought
+
+    def send_data(self, data):
         """This function sends the data to notion database"""
-        r = requests.post(url = self.url_add_pages, headers = self.headers, data = json.dumps(data))
+        r = requests.post(
+            url=self.url_add_pages,
+            headers=self.headers,
+            data=json.dumps(data),
+            timeout=10,
+        )
+        return r.status_code
```

Comment:
To potencjalnie niebezpieczne, nie widziałem żebyś sprawdzał status code == 200 dla tej funckji 
notion_book_databse.send_data(data) np. nie było.
Teraz założenie, jest że zawsze będzie OK, ale jak coś pójdzie nie tak to nawet użytkownik nie zauważy.

Powinno być sprawdzanie status_code tutaj lub wszędzie gdzie użyte i chociaż logowanie błędu. 
W bardziej skomplikowanych przypadkach retry, ale to zalezy od use case

Reply:
Valid point, dodam jakaś logike do okoła kodu. Rzeczywiście założyłem żę zawsze sie uda bo nie ma co sie nie udać, ale w sumie jak sie headersy zmienia w notion itd. to who knows

Reply:
done